File: backend/tpdb-backend-mono/src/data/park/entities/park.entity.ts
import { Resort } from '../../resort/entities/resort.entity';
import {
  Column,
  Entity,
  JoinColumn,
  ManyToOne,
  OneToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { ParkType } from '../../types/park-type/entities/park-type.entity';
import { User } from '../../../authentication/user/entities/user.entity';

@Entity()
export class Park {
  @PrimaryGeneratedColumn('uuid')
  id?: string;

  @Column({ nullable: true })
  createdAt: Date;

  @ManyToOne(() => User, { eager: true })
  @JoinColumn({ name: 'createUserId' })
  createdBy: User;

  @ManyToOne(() => User, { eager: true })
  @JoinColumn({ name: 'updateUserId' })
  updatedBy: User;

  @Column({ nullable: true })
  updatedAt: Date;
  @Column()
  name: string;

  @Column()
  description: string;

  @Column()
  openingDate: Date;

  @Column({ nullable: true })
  closingDate: Date;

  @OneToOne(() => ParkType)
  @JoinColumn()
  parkType: ParkType;

  @ManyToOne(() => Resort, (resort) => resort.parks)
  resort: Resort;
}

File: backend/tpdb-rest-backend/src/main/java/com/tpdb/domain/tpdbrestbackend/persistence/entities/data/ParkEntity.java
package com.tpdb.domain.tpdbrestbackend.persistence.entities.data;


import com.tpdb.domain.data.enums.ParkStatus;
import com.tpdb.domain.data.enums.types.ParkType;
import com.tpdb.domain.internal.scraper.enums.ScrapeSource;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(
        name = "park",
        uniqueConstraints = @UniqueConstraint(columnNames = {"source", "source_id"})
)
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class ParkEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private LocalDateTime created;
    private LocalDateTime updated;
//    private User updatedBy;
//    private User createdBy;

    private String name;
    @Column(columnDefinition = "text")
    private String description;
    @Enumerated(EnumType.STRING)
    private ParkType parkType;
    private LocalDate opening;
    private LocalDate closing;

    @Enumerated(EnumType.STRING)
    private ParkStatus status;
    private String address;
    private double areaSize;

    @Enumerated(EnumType.STRING)
    private ScrapeSource source;

    @Column(name="source_id")
    private String sourceId;
}
